core/chains/evm/headtracker/head_listener_test.go

```diff
@@ -168,7 +168,7 @@ func Test_HeadListener_SubscriptionErr(t *testing.T) {
 			cfg := configtest.NewGeneralConfig(t, nil)
 			evmcfg := evmtest.NewChainScopedConfig(t, cfg)
 			chStop := make(chan struct{})
-			hl := headtracker.NewEVMHeadListener(l, ethClient, headtracker.NewWrappedConfig(evmcfg), chStop)
+			hl := headtracker.NewHeadListener(l, ethClient, headtracker.NewWrappedConfig(evmcfg), chStop)
```

Comment:
There isn't other packages on heads, but there is a file for `heads`. vscode://eamodio.gitlens/link/r/559ba670360b80aba2811c6038309cc8639d0b75?url=https%3A%2F%2Fgithub.com%2Fsmartcontractkit%2Fchainlink.git

My concern is that using package `heads` might give the wrong idea of `heads` instead of them being the services that makes use of heads?

Reply:
Interesting idea!
Would it make sense to use the same naming pattern with the `txmgr` package for consistency?
Since it also has the pattern of `txmgr.txmgr`, `txmgr.broadcaster` , etc..

Reply:
@mohamed-mehany do you mean since the overall package is headtracker, it would be better to name it as `headtracker`? correct me if i am wrong, i think right now it has the same pattern as txmgr?

Reply:
Yes, right now they have "almost" similar naming pattern.
if I'm not mistaken, we have:
```
func txmgr.NewTxm(...) *Txm
func txmgr.NewBroadcaster(...) *Broadcaster
func txmgr.NewConfirmer(...) *Confirmer
.
.
``` 

My suggestion would be to use sth like:
```
func headtracker.NewHeadTracker(...) *HeadTracker
func headtracker.NewListener(...) *Listener
func headtracker.NewBroadcaster(...) *Broadcaster
.
.
``` 
Mainly to maintain consistency in naming. Just a minor nit though.

Reply:
IMHO we should not be targeting consistency with the old types. All of the long, redundant naming is not idiomatic and this is our opportunity to make it clear and readable before the API is locked in for third party integrations.

Reply:
I'm ok with both. I think I'm biased towards more verbose names due to my Java background :)